### PR TITLE
Api impl

### PR DESCRIPTION
**Pull request by incognitjoe**
Added a Falcon API for /heatmap/ and /results/, /results/{project}.

Spotted a problem with sessions in PipelineRun - the same session was global, so GETs running during INSERTs would break everything. Also became unbound if left idle for long enough, requiring an app restart, whoops!

### CODE REVIEW

File: cloudfoundry.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+def get_pipeline_data(pipelinebranch='develop', frequency=3600):
+    """
+    Retrieve pipeline data from Jenkins and persist it.
+    :param pipelinebranch: Branch to retrieve data about.
+    :param frequency: time in seconds to wait between fetches.
+    """
+    while True:
+        djinn.get_all_pipeline_results_and_save_to_db(pipelinebranch=pipelinebranch)
+        sleep(frequency)
```
Comment (wayofthepie) [inline cloudfoundry.py]: Why use sleep and not have this as a scheduled task?

File: djinn/__init__.py [hunk 2]
```diff
@@ -12,13 +15,23 @@ def __init__(self, jenkinsurl=None, dburl=None):
...
         self.logger = get_named_logger('Djinn')
         self.dj = DJenkins(url=jenkinsurl, logger=self.logger)
+        self.dburl = dburl
         self.db = PipelineResults(connection_url=dburl, echo=False)
+        self.service = AnalysisService(analysis=Analysis(), pipeline=self.db)
 
     def get_all_pipeline_results_and_save_to_db(self, pipelinebranch):
         """
         Fetch all repository pipeline data and write to database.
+        Could be memory-hungry depending on your instance - may be better to break it down by folder if so.
```
Comment (wayofthepie) [inline djinn/__init__.py]: Curious, does the API support paging?

Comment (incognitjoe) [inline djinn/__init__.py]: Jenkins? Not the way this method works - it's actually hitting the API separately for every repo and collating the results. Could be rewritten to perform writes to the db per repo by more or less copying the function this is calling.

For reference, it didn't even blink at our test instance - mem usage has been lower than 50MB constantly. If you had thousands of repos it'd be a problem, possibly.

File: cloudfoundry.py (hunk 1 above)
Comment (incognitjoe) [inline cloudfoundry.py]: Scheduled tasks are my next thing to experiment with actually(need to actually go through the Schedule docs, or look at Celery/MQ tasks instead), this is functional enough to start populating data regularly until I get around to it though.

**Pull request merged**